prometheus-net/Counter.cs

```diff
@@ -26,37 +26,27 @@ public void Inc(double increment = 1)
 
         public class Child : Advanced.Child, ICounter
         {
-            private double _value;
+            private ThreadSafeDouble _value;
 
             protected override void Populate(Metric metric)
             {
                 metric.counter = new Advanced.DataContracts.Counter();
                 metric.counter.value = Value;
             }
 
-            public void Inc(double increment = 1)
+            public void Inc(double increment = 1.0D)
             {
-                if (increment < 0)
-                {
-                    throw new InvalidOperationException("Counter cannot go down");
-                }
+                if (increment <= 0.0D)
+                    throw new InvalidOperationException("Counter can only go up");
 
-                double newCurrentValue = 0;
-                while (true)
```

Comment:
This implementation was not quite right. 
It would cause double computations on everything exception the first increment.
The new logic is encapsulated in ThreadSafeDouble and follows MSDN recommendations
https://msdn.microsoft.com/en-us/library/cd0811yf(v=vs.110).aspx